tests/run-macros/i17105/Macro_1.scala

```diff
@@ -0,0 +1,44 @@
+import scala.quoted.*
+
+inline def testExpr(inline body: Any) = ${ testExprImpl('body) }
+def testExprImpl(body: Expr[Any])(using Quotes): Expr[String] =
+  body match
+    case '{ def g(y: String) = "placeholder" + y; $a(g): String } =>
+      '{ $a((z: String) => s"[1st case] ${z}") }
+    case '{ def g(y: String)(z: String) = "placeholder" + y; $a(g): String } =>
+      '{ $a((z1: String) => (z2: String) =>  s"[2nd case] ${z1}, ${z2}") }
+    case '{ given i: String = "given"; def g(using s: String) = "placeholder"; $a(g, i): String } =>
+      '{ $a((s: String) => s"[3rd case] ${s}")(0) }
+    /* TODO issue-17105: This causes type error due to another issue from issue-17105 */
+    // case '{ def g(using s: String) = "placeholder"; $a(g) } =>
```

Comment:
With this test case, I got the following error.

```
-- [E172] Type Error: /home/yuito/code/dotty-workspace/issue-17105/macro_1.scala:39:56
39 |    case '{ def g(using s: String) = "placeholder"; $a(g): String } =>
   |                                                        ^
   |  No given instance of type String was found for parameter s of method g
```

This seems to be undesirable behavior considering that we don't necessarily resolve
contextual parameters at this point. If it's an issue, it should be fixed as another issue from this issue.

Reply:
We should fix that in another PR.

Reply:
Note: I created another issue https://github.com/lampepfl/dotty/issues/17905